### PR TITLE
Enabling further C++ strict conformance (/permissive-)

### PR DESCRIPTION
**Pull request by ahimberg**
add permissive- to our options in warnings.props so stricter C++ conformance is enabled by default.  Previously we only had it enabled in the ReactUWP folder.
With a few related build config changes:
* /cx projects require 2phase turned off, our sample app and test project have cx enabled right now
* turned off for the Desktop folder, @JunielKatarn to followup on fixing the boost socket integration

The fixes needed were mostly in test code.
I brought in one fix from my VS2019 pull request that is still in progress.

Also this adds a warning disable in msoFolly.h requested by office engineering working on compiler upgrades and hitting this warning in folly usage.

### CODE REVIEW

File: vnext/PropertySheets/Warnings.props [hunk 1]
```diff
@@ -28,6 +28,8 @@
 
   <ItemDefinitionGroup>
     <ClCompile>
+      <!-- /permissive- by default to enforce standards conformance, unless ENABLEPermissive has been set -->
+      <AdditionalOptions Condition="'$(ENABLEPermissive)' == ''">/permissive- %(AdditionalOptions)</AdditionalOptions>
```
Comment (JunielKatarn) [inline vnext/PropertySheets/Warnings.props]: Why is this in `Warnings.props`?

File: vnext/ReactUWP/ReactUWP.vcxproj [hunk 2]
```diff
@@ -81,7 +81,7 @@
       <PreprocessorDefinitions>REACTWINDOWS_BUILD;RN_PLATFORM=uwp;USE_EDGEMODE_JSRT;NOMINMAX;FOLLY_NO_CONFIG;RN_EXPORT=;JSI_EXPORT=;WIN32=0;WINRT=1;NOJSC;_HAS_AUTO_PTR_ETC;%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <AdditionalIncludeDirectories>$(ReactNativeWindowsDir);$(ReactNativeWindowsDir)Pch;$(ReactNativeWindowsDir)ReactUWP\GeneratedWinmdHeader;$(ReactNativeWindowsDir)ReactWindowsCore;$(ReactNativeWindowsDir)include\ReactWindowsCore;$(ReactNativeWindowsDir)include\ReactUWP;$(YogaDir);$(ReactNativeDir)\ReactCommon;$(ReactNativeWindowsDir)include;$(ReactNativeWindowsDir)stubs;$(ReactNativeWindowsDir)Shared;$(FollyDir);%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
       <LanguageStandard>stdcpp17</LanguageStandard>
-      <AdditionalOptions>/await /permissive- %(AdditionalOptions)</AdditionalOptions>
+      <AdditionalOptions>/await %(AdditionalOptions)</AdditionalOptions>
```
Comment (JunielKatarn) [inline vnext/ReactUWP/ReactUWP.vcxproj]: Why not use the `ENABLEPermissive` property you defined instead?

File: vnext/PropertySheets/Warnings.props (hunk 1 above)
Comment (ahimberg) [reply vnext/PropertySheets/Warnings.props]: enabling strict C++ warnings seemed like it should go with warnings?

File: vnext/ReactUWP/ReactUWP.vcxproj (hunk 2 above)
Comment (ahimberg) [reply vnext/ReactUWP/ReactUWP.vcxproj]: the trailing minus is the important part.  we are setting permissive OFF as the default.  You set ENABLEpermissive to turn off the default and get permissive msvc legacy C++ compat.  Once you fix desktop we'll remove the ENABLEpermissive setting so - is enabled everywhere

**Pull request merged**